src/NavigatorBackendType.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Stores the view models and interface types between the UI layer and its
+ * backend.
+ */
+
+/**
+ * A GitHub Pull Request, GitLab Merge Request, or similar.
+ */
+export interface PullRequestInfo {
+  url: string;
+  shortName: string;
+
+  isOutdated: boolean;
+}
+
+export interface Commit {
+  name: string;
+  hash: string;
+  timestamp: string;
+  author: string;
+  title: string;
+  branchNames: string[];
+  pullRequestInfo?: PullRequestInfo;
+
+  parentCommits: Commit[];
+  childCommits: Commit[];
+}
+
+export interface Repository {
+  path: string;
+  hasUncommittedChanges: boolean;
+  headHash: string;
+
+  /** The earliest interesting commit. */
+  rootDisplayCommit: Commit;
+}
+
+export type NavigatorBackend = {
+  // Getters
+  /**
+   * Asynchronously gets repository information.
+   * @param repoPath Path to repository root.
+   * @returns Cached repository information, along with a promise that returns
+   * the repository information updated with remote data (e.g. PRs).
+   */
+  getRepositoryInformation(
+    repoPath: string,
+  ): Promise<{
+    repo: Repository;
```

Comment:
Should we not fetch all the branch names too?

Reply:
I attached the branch names to the individual commits as I don't think we need to know the names of all the branches at the repo level. Do you think that makes sense?

Reply:
Yeah yeah, sorry, each commit can be a feature branch, makes sense. LGTM!

Reply:
Or multiple branches too, e.g. origin/feat/some-branch, feat/some-branch-1, etc